packages/@aws-cdk-testing/framework-integ/test/aws-dynamodb/test/integ.dynamodb.policy.ts

```diff
@@ -0,0 +1,47 @@
+import { App, RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+const app = new App();
+
+class TestStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // table with resource policy
+    new dynamodb.Table(this, 'TableTest1', {
+      partitionKey: {
+        name: 'id',
+        type: dynamodb.AttributeType.STRING,
+      },
+      removalPolicy: RemovalPolicy.DESTROY,
+      resourcePolicy: iam.PolicyDocument.fromJson({
+        Statement: [
+          {
+            Action: 'dynamodb:*',
+            Effect: 'Allow',
+            Principal: {
+              AWS: '123456789101',
+            },
+            Resource: '*',
+          },
+        ],
+        Version: '2012-10-17',
+      }),
+    });
+
+    // table without resource policy
+    new dynamodb.Table(this, 'TableTest2', {
+      partitionKey: {
+        name: 'id',
+        type: dynamodb.AttributeType.STRING,
+      },
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+}
+
+new TestStack(app, 'ResourcePolicyTest');
+
+app.synth();
```

Comment:
Can you run this with `yarn integ-runner...` ? It should produce a snapshot.

Reply:
I can. I was just running `cdk synth`. So `yarn integ-runner filename.js` is the way to run it?

Reply:
yeah, take a look here: https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md